pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -335,6 +364,54 @@ func (p *PriorityQueue) Run() {
 	go wait.Until(p.flushUnschedulablePodsLeftover, 30*time.Second, p.stop)
 }
 
+// runRequeuePlugins calls Requeue() of only plugins registered in pInfo.unschedulablePlugins.
+// If more than one Requeue() returns success, the scheduling queue enqueues the Pod to activeQ or backoffQ.
+// If all Requeue() returns unschedulable, the scheduling queue enqueues the Pod back to unschedulable Pod pool
+// because no plugin changes the scheduling result via the event.
+func (p *PriorityQueue) runRequeuePlugins(ctx context.Context, pInfo *framework.QueuedPodInfo, event *framework.ClusterEvent, oldObj, newObj interface{}) bool {
+	if pInfo.UnschedulablePlugins.Len() == 0 {
```

Comment:
How is access to `pInfo.UnschedulablePlugins` protected against concurrent access?

`runRequeuePlugins` may get invoked in `movePodsToActiveOrBackoffQueue` while some goroutine is scheduling the pod. `runRequeuePlugins` holds the scheduling queue's mutex, but that other goroutine doesn't.

Reply:
Great catch. Let me think about it.